src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
+        capacity = table.length;
+        int position = getPosition(key, capacity);
+        threshold = (int) (LOAD_FACTOR * capacity);
+
+        if (size >= threshold) {
+            table = resize();
+        }
+
+        for (int i = 0; i < capacity; i++) {
+            if (table[i] != null && Math.abs(table[i].key == null ? 0
+                    : table[i].hash % capacity) == position) {
+                Node<K, V> node = table[i];
+                while (node != null) {
+                    if (Objects.deepEquals(node.key, key)) {
+                        node.value = value;
+                        return;
+
+                    } else if (node.next == null) {
+                        node.next = newNode;
+                        size++;
+                        return;
+                    }
+                    node = node.next;
+                }
+            }
+        }
 
+        table[position] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (capacity == 0) {
+            return null;
+        }
+        int position = getPosition(key, capacity);
+        for (int i = 0; i < capacity; i++) {
+            if (table[i] != null && i == position) {
+                Node<K, V> node = table[i];
+                for (int j = 0; j <= size; j++) {
+                    if (node.key == null && key == null || Objects.equals(node.key, key)) {
+                        return node.value;
+                    } else {
+                        if (node.next != null) {
+                            node = node.next;
+                        }
+                    }
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V>[] resize() {
+        if (threshold == MAX_CAPACITY) {
+            return table;
+        }
+
+        int newCapacity = capacity * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        threshold = (int) (LOAD_FACTOR * newCapacity);
+
+        for (int i = 0; i < table.length; i++) {
+            Node<K, V> currentNode = table[i];
+            while (currentNode != null) {
+                int position = getPosition(currentNode.key, newCapacity);
+                Node<K, V> newNode = newTable[position];
+                if (newNode == null) {
+                    newTable[position] = new Node<>(currentNode.hash,
+                            currentNode.key, currentNode.value, null);
+                } else {
+                    newNode.next = new Node<>(currentNode.hash,
+                            currentNode.key, currentNode.value, newNode.next);
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        return newTable;
+    }
+
+    private int getPosition(K key, int capacity) {
+        return Math.abs(key == null ? 0
+                : key.hashCode() % capacity);
```

Comment:
why do you need `capacity` param? you can use `table.length` instead.

Reply:
Because table.length() is changed only in put, but not in the resize. I'll delete second parametr and leave onle key